app/notification.go

```diff
@@ -249,7 +249,7 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 					Following:             true,
 					IncrementMentions:     incrementMentions,
 					UpdateFollowing:       updateFollowing,
-					UpdateViewedTimestamp: userID == post.UserId,
+					UpdateViewedTimestamp: false,
```

Comment:
It seems like we are setting it to false here, and then again making another call later if `uid == post.UserId`. Can you clarify why is that?

Reply:
Having this made previous unread replies and mentions for the commenter be always 0, so I moved the "marking the thread as viewed" after getting previous unread counts.